### PR TITLE
Initialize Testing

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Einen Test für den ProjectService erstellen](https://trello.com/c/5j5hoTtP/178-einen-test-f%C3%BCr-den-projectservice-erstellen)

**Comment by ChristianBeilschmidt** [comment]
[Einen Test für den ProjectService erstellen](https://trello.com/c/5j5hoTtP/178-einen-test-f%C3%BCr-den-projectservice-erstellen)

File: projects/wave-core-new/src/lib/project/project.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    mapServiceSpy = jasmine.createSpyObj('MapService', ['getViewportSizeStream']);
+    backendSpy = jasmine.createSpyObj('BackendService', ['createProject', 'listProjects']);
+    userServiceSpy = jasmine.createSpyObj('UserService', ['getSessionStream', 'getSessionTokenForRequest']);
+
+    // always return the same session
+    userServiceSpy.getSessionStream.and.returnValue(
+        of<Session>({
+            sessionToken: 'session-token',
+            user: new User({
+                id: 'user-id',
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/project/project.service.spec.ts:35]: maybe use an actual valid UUID?

File: projects/wave-core-new/src/lib/project/project.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            }),
+            validUntil: moment.utc('2020-01-01'),
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/project/project.service.spec.ts:37]: is it a good idea to use a date in the past? It basically means the session is invalid

File: projects/wave-core-new/src/lib/project/project.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        }),
+    );
+    userServiceSpy.getSessionTokenForRequest.and.returnValue(of<UUID>('session-token'));
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/project/project.service.spec.ts:40]: maybe use an actual valid UUID?

File: projects/wave-core-new/src/lib/project/project.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        (userServiceSpy as any) as UserService,
+    );
+});
+
+it(
+    'should create a default project',
+    waitForAsync(() => {
+        backendSpy.createProject.and.returnValue(
+            of<CreateProjectResponseDict>({
+                id: 'the-project-id',
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/project/project.service.spec.ts:59]: maybe use an actual valid UUID?

File: projects/wave-core-new/src/lib/project/project.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            }),
+        );
+
+        projectService.createDefaultProject().subscribe(
+            (project) =>
+                expect(project.toDict()).toEqual(
+                    new Project({
+                        id: 'the-project-id',
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/project/project.service.spec.ts:67]: maybe use an actual valid UUID?

File: projects/wave-core-new/src/lib/project/project.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                        layers: [],
+                        timeStepDuration: {
+                            durationAmount: 1,
+                            durationUnit: 'month',
+                        },
+                    }).toDict(),
+                ),
+            (error) => fail(error),
+            () => {
+                expect(backendSpy.createProject).toHaveBeenCalledOnceWith(
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/project/project.service.spec.ts:82]: this doesn't work for me. If I don't call the createDefaultProject method, this test still doesn't fail.

File: projects/wave-core-new/src/lib/project/project.service.spec.ts (hunk 5 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/project/project.service.spec.ts]: 2e6ebe9

File: projects/wave-core-new/src/lib/project/project.service.spec.ts (hunk 4 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/project/project.service.spec.ts]: 2e6ebe9

File: projects/wave-core-new/src/lib/project/project.service.spec.ts (hunk 3 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/project/project.service.spec.ts]: 2e6ebe9

File: projects/wave-core-new/src/lib/project/project.service.spec.ts (hunk 1 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/project/project.service.spec.ts]: 2e6ebe9

File: projects/wave-core-new/src/lib/project/project.service.spec.ts (hunk 2 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/project/project.service.spec.ts]: ae20f9d

File: projects/wave-core-new/src/lib/project/project.service.spec.ts (hunk 6 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/project/project.service.spec.ts]: 23f478a

**Pull request merged**